code/solutions/nested-stacks/main.yaml

```diff
@@ -6,7 +6,7 @@ Parameters:
   S3BucketName:
     Description: S3 bucket name for the Nested Stacks. S3 bucket name can include numbers, lowercase letters, uppercase letters, and hyphens (-). It cannot start or end with a hyphen (-).
     Type: String
-    AllowedPattern: ^[0-9a-zA-Z]+([0-9a-zA-Z-.]*[0-9a-zA-Z])*$
+    AllowedPattern: ^(?!(^xn--|.+-s3alias$))^[a-z0-9][a-z0-9-]{1,61}[a-z0-9]$
```

Comment:
What's the difference between the two new patterns? When is one used over the other?

Reply:
The new pattern fixes an issue where we allowed capital case. Also the new pattern complies with [Bucket naming rules](https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html)

Reply:
Oh, I misunderstood your question. Correct, you have answered it. The prefix is and full bucket name have two different patterns :)